__tests__/document-collection.test.js

```diff
@@ -0,0 +1,75 @@
+jest.mock('../lib/files.js', () => ({
+  readFile: jest.fn(),
+  writeFile: jest.fn(),
+  readdir: jest.fn(),
+}));
+
+// for setting up mock expectations
+const { readFile, writeFile, readdir } = require('../lib/files');
+const DocumentCollection = require('../lib/document-collection');
+const documentNew = new DocumentCollection('document');
+
+describe('Document Collection', () => {
+
+  const exampleObject = { name: 'Alex' };
+
+  it('saves a file', () => {
+    // arrange
+    writeFile.mockResolvedValue(exampleObject);
+    
+    // act
+    return documentNew.save(exampleObject)
+      .then(() => {
+        const dest = `./document/${exampleObject.id}.json`;
```

Comment:
Nice job with setting up your dest test - I would have never thought to do exampleObject.id after the function has been called, but it totally works since the function is changing the original object and not making a new one. Awesome work!!